src/styles/builders.js

```diff
@@ -584,6 +584,67 @@ Builders.buildQuadsForPoints = function (
     }
 };
 
+// Build a billboard sprite quad centered on a point. Sprites are intended to be drawn in screenspace, and have
+// properties for width, height, angle, and a scale factor that can be used to interpolate the screenspace size
+// of a sprite between two zoom levels.
+Builders.buildSpriteQuadsForPoints = function (
+    points,
+    width, height, angle, scale,
+    vertex_data, vertex_template,
+    scaling_index,
+    { texcoord_index, texcoord_scale }) {
+
+    let w2 = width / 2;
+    let h2 = height / 2;
+    let scaling = [
+        [-w2, -h2],
```

Comment:
Just a reminder that if we need, we can add an `anchor` property that will make the sprite/label rotate around a position other than the center point. That can be added later though.